### ISSUE TITLE
Issue with bloom models

### ISSUE THREAD

**Issue by charlesmartin14** [CalculatedContent/WeightWatcher#221]
**When trying to analyze the very large BLOOM models, we get the error**

in extract_pytorch_statedict()

```
model_dir = cache_dir
config = ww.WeightWatcher.process_pytorch_bins(model_dir=model_dir, model_name='bloom-big')
ERROR:weightwatcher:Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/weightwatcher/weightwatcher.py", line 4437, in process_pytorch_bins
    layer_configs = WeightWatcher.extract_pytorch_statedict(weights_dir, model_name, state_dict_filename, start_id)
  File "/usr/local/lib/python3.9/dist-packages/weightwatcher/weightwatcher.py", line 4339, in extract_pytorch_statedict
    W = T.cpu().detach().numpy() # do we need  ?
TypeError: Got unsupported ScalarType BFloat16

CRITICAL:weightwatcher:Unknown problem, stopping
```

**Comment by charlesmartin14** [comment]
**ChatGPT suggests**


As a Python programmer, I can see that the issue here is that the Torch Tensor T has a datatype of BFloat16, which is not supported by the numpy library for conversion to a numpy array using the numpy() function.

The numpy() function is used to convert a PyTorch Tensor to a NumPy array. However, not all PyTorch datatypes are supported by numpy(), and BFloat16 is one such datatype. Therefore, attempting to convert a BFloat16 Tensor to a NumPy array using numpy() will result in a TypeError as seen in the code snippet provided.

To resolve this issue, you can first convert the BFloat16 Tensor to a supported datatype, such as Float32, using the float() method. For example, the code can be modified as follows:


`W = T.cpu().detach().float().numpy()`

This should allow the code to execute without errors and convert the PyTorch Tensor T to a NumPy array W with datatype Float32.

_IDK if this works or not...need to check_

**Comment by charlesmartin14** [comment]
This has been fixed in version 0.6.7.7
